Issues with Yandex (old/new designs).

**Describe the bug**
New design ofYandex Music web version doesn't scrobble. 

**How to reproduce**
Steps to reproduce the behavior:
1. Go to '(https://music.yandex.ru/)'
2. Click on play 
3. Doesnt scrobble 

**Expected behavior**
Expected scrobble

By the way, @jaccarmac, when I synced my local branch with master branch of web scrobbler, i started to suffer from problem with src/core/storage/wrapper.ts. 
Lints fails because of it. idk where the problem exactly, but this is it. I tried to fix it locally and lints passed, but idk how it will work on build.

```
/home/bulat/web-scrobbler/src/core/storage/wrapper.ts
  199:5   error  'StorageAreaWithUsage' is an 'error' type that acts as 'any' and overrides all other types in this union type  @typescript-eslint/no-redundant-type-constituents
  219:6   error  'StorageAreaWithUsage' is an 'error' type that acts as 'any' and overrides all other types in this union type  @typescript-eslint/no-redundant-type-constituents
  223:3   error  Unsafe assignment of an error typed value                                                                      @typescript-eslint/no-unsafe-assignment
  245:9   error  Unsafe assignment of an error typed value                                                                      @typescript-eslint/no-unsafe-assignment
  245:22  error  Unsafe call of a(n) error type typed value                                                                   @typescript-eslint/no-unsafe-call
  245:35  error  Unsafe member access .get on an error typed value                                                            @typescript-eslint/no-unsafe-member-access
  247:16  error  Unsafe member access [this.namespace] on an error typed value                                                @typescript-eslint/no-unsafe-member-access
  288:9   error  Unsafe call of a(n) error type typed value                                                                   @typescript-eslint/no-unsafe-call
  288:22  error  Unsafe member access .set on an error typed value                                                            @typescript-eslint/no-unsafe-member-access
  346:9   error  Unsafe call of a(n) error type typed value                                                                   @typescript-eslint/no-unsafe-call
  346:22  error  Unsafe member access .remove on an error typed value                                                         @typescript-eslint/no-unsafe-member-access

✖️ 11 problems (11 errors, 0 warnings)
```

--- a/src/connectors/yandex-music.ts
+++ b/src/connectors/yandex-music.ts
@@ -17,6 +17,7 @@ function setupConnector(): void {
 }
 
 function setupOldConnector(): void {
+	Util.debugLog('setting up for Yandex Music old design');
 	const observer = new MutationObserver(() => {
 		const el = document.querySelector('.track.track_type_player');
 		if (el) {
@@ -95,150 +96,67 @@ function setupOldConnector(): void {
 // NEW CONNECTOR
 
 function setupNewConnector(): void {
-	Connector.playerSelector =
-		"section[class*='PlayerBarDesktopWithBackgroundProgressBar_root']";
+	Util.debugLog('setting up for Yandex Music new design');
 
-	Connector.getTrack = (): string | null => {
-		const playerContainer = document.querySelector(
-			'section[class*="PlayerBarDesktopWithBackgroundProgressBar_root"]',
-		);
-		if (!playerContainer) {
-			return null;
+	// this accommodates both desktop and mobile player.
+	// during fullscreen the bar still exists underneath
+	const playerSelector = 'section[class*="PlayerBar_root__"]';
+	Connector.playerSelector = playerSelector;
+
+	Connector.getTrack = () => {
+		const titleContainer = `${playerSelector} [class*="Meta_titleContainer__"]`;
+		let title = Util.getTextFromSelectors(
+			`${titleContainer} [class*="Meta_title__"]`,
+		)?.trim();
+		const version = Util.getTextFromSelectors(
+			`${titleContainer} [class*="Meta_version__"]`,
+		)?.trim();
+
+		if (version) {
+			title += ` (${version})`;
 		}
 
-		const titleContainer = playerContainer.querySelector(
-			'div[class*="Meta_titleContainer"]',
-		);
-		if (!titleContainer) {
-			return null;
-		}
-
-		const link = titleContainer.querySelector('a');
-		if (!link) {
-			return null;
-		}
-
-		const titleSpan = link.querySelector('span[class*="Meta_title__"]');
-		if (!titleSpan) {
-			return null;
-		}
-
-		let trackName = titleSpan.textContent?.trim() ?? '';
-
-		const versionSpan = link.nextElementSibling as HTMLElement | null;
-		if (versionSpan && versionSpan.className.includes('Meta_version__')) {
-			const versionText = versionSpan.textContent
-				?.replace(/\u00a0/g, ' ')
-				.trim();
-			if (versionText) {
-				trackName += ` (${versionText})`;
-			}
-		}
-
-		return trackName || null;
+		return title;
 	};
 
-	Connector.getArtist = (): string | null => {
-		const playerContainer = document.querySelector(
-			'section[class*="PlayerBarDesktopWithBackgroundProgressBar_root"]',
+	Connector.getArtist = () => {
+		return Util.getTextFromSelectors(
+			`${playerSelector} [class*="Meta_artists__"]`,
 		);
-		if (!playerContainer) {
-			return null;
-		}
-
-		const artistContainer = playerContainer.querySelector(
-			'div[class*="SeparatedArtists_root"]',
-		);
-		if (!artistContainer) {
-			return null;
-		}
-
-		const links = artistContainer.querySelectorAll('a');
-		const artists: string[] = [];
-
-		links.forEach((a) => {
-			const span = a.querySelector('span[class*="Meta_artistCaption"]');
-			if (span?.textContent?.trim()) {
-				artists.push(span.textContent.trim());
-			}
-		});
-
-		return artists.length ? artists.join(', ') : null;
 	};
 
-	Connector.getTrackArt = (): string | null => {
-		const trackArtImage: HTMLImageElement | undefined = [
-			'img[class*="PlayerBarMobile_cover"]',
-			'img[class*="PlayerBarDesktopWithBackgroundProgressBar_cover"]',
-			'img[class*="FullscreenPlayerDesktopPoster_cover"]',
-		]
-			.map((cover) => document.querySelector<HTMLImageElement>(cover))
-			.find((cover) => cover !== null);
+	Connector.getTrackArt = () => {
+		const url = Util.extractImageUrlFromSelectors('img[class*="_cover__"]');
 
-		if (trackArtImage === undefined) {
-			return null;
-		}
-
-		const url = new URL(
-			trackArtImage.src,
-			window.location.origin,
-		).toString();
-
-		return url.replace(/\d+x\d+/, '800x800');
+		return url?.replace(/\d+x\d+$/g, '800x800');
 	};
 
 	Connector.getCurrentTime = (): number | null => {
-		const el = document.querySelector('[class*="Timecode_root_start"]');
-		const parts = el?.textContent?.trim().split(':');
-		if (!parts || parts.length !== 2) {
-			return null;
-		}
-
-		const minutes = parseInt(parts[0], 10);
-		const seconds = parseInt(parts[1], 10);
-		if (isNaN(minutes) || isNaN(seconds)) {
-			return null;
-		}
-
-		return minutes * 60 + seconds;
+		const timeStr = Util.getAttrFromSelectors(
+			`${playerSelector} input[class*=_slider__]`,
+			'value',
+		);
+		return timeStr ? parseFloat(timeStr) : null;
 	};
 
 	Connector.getDuration = (): number | null => {
-		const el = document.querySelector('[class*="Timecode_root_end"]');
-		const parts = el?.textContent?.trim().split(':');
-		if (!parts || parts.length !== 2) {
-			return null;
-		}
-
-		const minutes = parseInt(parts[0], 10);
-		const seconds = parseInt(parts[1], 10);
-		if (isNaN(minutes) || isNaN(seconds)) {
-			return null;
-		}
-
-		return minutes * 60 + seconds;
+		const durStr = Util.getAttrFromSelectors(
+			`${playerSelector} input[class*=_slider__]`,
+			'max',
+		);
+		return durStr ? parseFloat(durStr) : null;
 	};
 
-	Connector.isPlaying = (): boolean => {
-		const buttons = Array.from(
-			document.querySelectorAll('button'),
-		) as HTMLElement[];
-		for (const btn of buttons) {
-			if (
-				Array.from(btn.classList).some((c) =>
-					c.includes('BaseSonataControlsDesktop_sonataButton'),
-				)
-			) {
-				const label = btn.getAttribute('aria-label');
-				if (label === 'Пауза') {
-					return true;
-				}
-				if (label === 'Воспроизведение') {
-					return false;
-				}
-			}
-		}
-		return false;
+	Connector.isPlaying = () => {
+		const playPauseButtonSelectors = [
+			`${playerSelector} button[class*="BaseSonataControlsDesktop_sonataButton__"] svg[class*="BaseSonataControlsDesktop_playButtonIcon__"] use`,
+			`${playerSelector} button[class*="SonataControls_root__"] svg use`,
+		];
+
+		return Util.getAttrFromSelectors(
+			playPauseButtonSelectors,
+			'xlink:href',
+		)?.includes('pause');
 	};
 
 	Connector.onStateChanged();

--- a/src/connectors/yandex-music.ts
+++ b/src/connectors/yandex-music.ts
@@ -1,174 +1,74 @@
 export {};
 
-setupConnector();
+// this accommodates both desktop and mobile player.
+// during fullscreen the bar still exists underneath
+const playerSelector =
+	'section[class*="PlayerBar_root__"]:has([class*="Meta_titleContainer__"])';
+Connector.playerSelector = playerSelector;
 
-function setupConnector(): void {
-	const body = document.querySelector('body');
-	const isNewDesign =
-		body?.classList.contains('ym-font-music') &&
-		(body.classList.contains('ym-dark-theme') ||
-			body.classList.contains('ym-light-theme'));
+Connector.getTrack = () => {
+	const titleContainer = `${playerSelector} [class*="Meta_titleContainer__"]`;
+	let title = Util.getTextFromSelectors(
+		`${titleContainer} [class*="Meta_title__"]`,
+	)?.trim();
+	const version = Util.getTextFromSelectors(
+		`${titleContainer} [class*="Meta_version__"]`,
+	)?.trim();
 
-	if (isNewDesign) {
-		setupNewConnector();
-	} else {
-		setupOldConnector();
-	}
-}
-
-function setupOldConnector(): void {
-	Util.debugLog('setting up for Yandex Music old design');
-	const observer = new MutationObserver(() => {
-		const el = document.querySelector('.track.track_type_player');
-		if (el) {
-			observer.disconnect();
-			Connector.playerSelector = '.track.track_type_player';
-		}
-	});
-
-	const btn = document.querySelector('.player-controls__btn_play');
-	if (btn) {
-		const btnObserver = new MutationObserver(() => {
-			Connector.onStateChanged();
-		});
-		btnObserver.observe(btn, {
-			attributes: true,
-			attributeFilter: ['class'],
-		});
+	if (version) {
+		title += ` (${version})`;
 	}
 
-	const trackObserver = new MutationObserver(() => {
-		Connector.onStateChanged();
-	});
+	return title;
+};
 
-	const trackNode = document.querySelector(
-		'.player-controls__track-container',
+Connector.artistSelector = `${playerSelector} [class*="Meta_artists__"]`;
+
+Connector.getTrackArt = () => {
+	const url = Util.extractImageUrlFromSelectors('img[class*="_cover__"]');
+
+	return url?.replace(/\d+x\d+$/g, '800x800');
+};
+
+Connector.getCurrentTime = () => {
+	const timeStr = Util.getAttrFromSelectors(
+		`${playerSelector} input[class*=_slider__]`,
+		'value',
 	);
-	if (trackNode) {
-		trackObserver.observe(trackNode, { childList: true, subtree: true });
-	}
+	return timeStr ? parseFloat(timeStr) : null;
+};
 
-	observer.observe(document.body, { childList: true, subtree: true });
+Connector.getDuration = () => {
+	const durStr = Util.getAttrFromSelectors(
+		`${playerSelector} input[class*=_slider__]`,
+		'max',
+	);
+	return durStr ? parseFloat(durStr) : null;
+};
 
-	Connector.trackSelector = '.track__title';
-	Connector.artistSelector = '.d-artists.d-artists__expanded';
+Connector.isPlaying = () => {
+	const playPauseButtonSelectors = [
+		`${playerSelector} button[class*="BaseSonataControlsDesktop_sonataButton__"] svg[class*="BaseSonataControlsDesktop_playButtonIcon__"] use`,
+		`${playerSelector} [class*=PlayerBarMobile_infoButtons__] button:last-child svg use`,
+	];
 
-	Connector.getTrackArt = (): string | null => {
-		const container = document.querySelector(
-			'.player-controls__track-container',
-		);
-		if (!container) {
-			return null;
-		}
+	// xlink:href is deprecated, they will probably change it out at some point
+	const useHrefAttr =
+		Util.getAttrFromSelectors(playPauseButtonSelectors, 'xlink:href') ??
+		Util.getAttrFromSelectors(playPauseButtonSelectors, 'href');
 
-		const images = container.querySelectorAll<HTMLImageElement>('img');
-		for (const img of images) {
-			const src = img.getAttribute('src');
-			if (src && src.includes('50x50')) {
-				const absoluteUrl = new URL(
-					src,
-					window.location.origin,
-				).toString();
-				return absoluteUrl.replace('50x50', '800x800');
-			}
-		}
-		return null;
-	};
+	return useHrefAttr?.includes('pause') ?? false;
+};
 
-	Connector.getCurrentTime = (): number | null => {
-		const el = document.querySelector('.progress__bar.progress__text');
-		const timeStr = el?.getAttribute('data-played-time');
-		return timeStr ? parseFloat(timeStr) : null;
-	};
+Connector.isLoved = () => {
+	const playPauseButtonSelectors = [
+		`${playerSelector} [class*="PlayerBarDesktopWithBackgroundProgressBar_sonata__"] button:first-child svg use`,
+		`${playerSelector} [class*=PlayerBarMobile_infoButtons__] button:first-child svg use`,
+	];
 
-	Connector.getDuration = (): number | null => {
-		const el = document.querySelector('.progress__bar.progress__text');
-		const durStr = el?.getAttribute('data-duration');
-		return durStr ? parseFloat(durStr) : null;
-	};
+	const useHrefAttr =
+		Util.getAttrFromSelectors(playPauseButtonSelectors, 'xlink:href') ??
+		Util.getAttrFromSelectors(playPauseButtonSelectors, 'href');
 
-	Connector.isPlaying = (): boolean => {
-		const btn = document.querySelector('.player-controls__btn_play');
-		return btn?.classList.contains('player-controls__btn_pause') ?? false;
-	};
-}
-
-// NEW CONNECTOR
-
-function setupNewConnector(): void {
-	Util.debugLog('setting up for Yandex Music new design');
-
-	// this accommodates both desktop and mobile player.
-	// during fullscreen the bar still exists underneath
-	const playerSelector = 'section[class*="PlayerBar_root__"]';
-	Connector.playerSelector = playerSelector;
-
-	Connector.getTrack = () => {
-		const titleContainer = `${playerSelector} [class*="Meta_titleContainer__"]`;
-		let title = Util.getTextFromSelectors(
-			`${titleContainer} [class*="Meta_title__"]`,
-		)?.trim();
-		const version = Util.getTextFromSelectors(
-			`${titleContainer} [class*="Meta_version__"]`,
-		)?.trim();
-
-		if (version) {
-			title += ` (${version})`;
-		}
-
-		return title;
-	};
-
-	Connector.getArtist = () => {
-		return Util.getTextFromSelectors(
-			`${playerSelector} [class*="Meta_artists__"]`,
-		);
-	};
-
-	Connector.getTrackArt = () => {
-		const url = Util.extractImageUrlFromSelectors('img[class*="_cover__"]');
-
-		return url?.replace(/\d+x\d+$/g, '800x800');
-	};
-
-	Connector.getCurrentTime = (): number | null => {
-		const timeStr = Util.getAttrFromSelectors(
-			`${playerSelector} input[class*=_slider__]`,
-			'value',
-		);
-		return timeStr ? parseFloat(timeStr) : null;
-	};
-
-	Connector.getDuration = (): number | null => {
-		const durStr = Util.getAttrFromSelectors(
-			`${playerSelector} input[class*=_slider__]`,
-			'max',
-		);
-		return durStr ? parseFloat(durStr) : null;
-	};
-
-	Connector.isPlaying = () => {
-		const playPauseButtonSelectors = [
-			`${playerSelector} button[class*="BaseSonataControlsDesktop_sonataButton__"] svg[class*="BaseSonataControlsDesktop_playButtonIcon__"] use`,
-			`${playerSelector} button[class*="SonataControls_root__"] svg use`,
-		];
-
-		return Util.getAttrFromSelectors(
-			playPauseButtonSelectors,
-			'xlink:href',
-		)?.includes('pause');
-	};
-
-	Connector.onStateChanged();
-
-	const playerNode = document.querySelector(Connector.playerSelector);
-	if (playerNode) {
-		const observer = new MutationObserver(() => Connector.onStateChanged());
-		observer.observe(playerNode, { childList: true, subtree: true });
-	} else {
-		console.warn(
-			'[Yandex Connector] Плеер не найден — используем fallback через setInterval',
-		);
-		setInterval(() => Connector.onStateChanged(), 1000);
-	}
-}
+	return useHrefAttr?.includes('liked');
+};

